astropy/cosmology/core.py

```diff
@@ -1326,7 +1329,7 @@ def angular_diameter_distance_z1z2(self, z1, z2):
         z1 = np.atleast_1d(z1)
         z2 = np.atleast_1d(z2)
 
-        if z1.size != z2.size:
+        if z1.size != z2.size and z1.size != 1 and z2.size !=1:
```

Comment:
Is there a reason not to just rely on numpy broadcasting? I'm a bit confused why one would bother with the test at all, but if it is felt it is needed, I would write it as:
```
try:
    np.broadcast(z1, z2)
except ValueError:
    raise ValueError(...)
```
But I would advocate just removing the test: the next `if` statement will raise a `ValueError` anyway if `z1` and `z2` have incompatible shapes.